python/pyspark/sql/column.py

```diff
@@ -304,7 +304,7 @@ def __ne__(  # type: ignore[override]
     |(value = foo)|(value <=> foo)|(value <=> NULL)|
     +-------------+---------------+----------------+
     |         true|           true|           false|
-    |         null|          false|            true|
+    |         NULL|          false|            true|
```

Comment:
is it consistent with inner struct fields?

Reply:
Do you mean as follows? It looks consistent with inner struct fields

spark-sql CLI
``` shell
spark-sql> select array(1, null);
[1,null]
```

df.show
```shell
spark.sql("select array(1,null)").show
+--------------+
|array(1, NULL)|
+--------------+
|     [1, null]|
+--------------+
```

Reply:
them it seems the current behavior is better? null value is always displayed as `null`, no matter it's top level column or inner struct field.

Reply:
Agree with you, but, seems `NULL` is more generic, like MySQL:
```shell
mysql> select null;
+------+
| NULL |
+------+
| NULL |
+------+
1 row in set (0.04 sec)

```

Reply:
OK, to keep consistent, it seems we should update `Cast` code when casting array/struct/map to string and the element/field is null.